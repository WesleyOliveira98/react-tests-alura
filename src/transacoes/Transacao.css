.transacao-container {
    width: 400px;
    margin: 2px auto;
    display: grid;
    grid-template-columns: 1fr 3fr 1fr;
    background-color: rgb(62,160,63,0.3);
}