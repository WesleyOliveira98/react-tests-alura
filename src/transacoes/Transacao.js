import React from 'react';
import PropTypes from 'prop-types';

import './Transacao.css'


function Transacao({ data, tipo, valor }) {
    return (
        <div className="transacao-container">
            <p>{data}</p>
            <p>{tipo}</p>
            <p>R$ {valor}</p>
        </div>
    )
};

Transacao.defaultProps = {
    tipo: '',
    valor: '0',
    data: '',
}

Transacao.propTypes = {
    tipo: PropTypes.string,
    valor: PropTypes.string,
    data: PropTypes.string,
}

export default Transacao;