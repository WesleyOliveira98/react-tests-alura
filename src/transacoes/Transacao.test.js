import React from 'react';
import { render } from '@testing-library/react';
import Transacao from './Transacao';

describe('Componente de transação de extrato', () => {
    it('O snapshot do componente deve permanecer sempre o mesmo', () => {
        const { container } = render(<Transacao 
            data='19/01/2022'
            tipo='saque'
            valor='20.00'
        />)

        expect(container.firstChild).toMatchSnapshot();
    })
})