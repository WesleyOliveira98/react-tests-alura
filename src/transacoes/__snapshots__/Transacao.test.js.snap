// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transação de extrato O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="transacao-container"
>
  <p>
    19/01/2022
  </p>
  <p>
    saque
  </p>
  <p>
    R$ 
    20.00
  </p>
</div>
`;
